src/instances/PS2AlertsMetagameInstance.ts

```diff
@@ -0,0 +1,78 @@
+import PS2AlertsInstanceAbstract from './PS2AlertsInstanceAbstract';
+import {MetagameEventType} from '../constants/metagameEventType';
+import {World} from '../constants/world';
+import {Zone} from '../constants/zone';
+import PS2AlertsInstanceInterface from '../interfaces/PS2AlertsInstanceInterface';
+import ApplicationException from '../exceptions/ApplicationException';
+import {Ps2alertsEventState} from '../constants/ps2alertsEventState';
+
+export default class PS2AlertsMetagameInstance extends PS2AlertsInstanceAbstract implements PS2AlertsInstanceInterface {
+    public readonly zone: Zone;
+    public readonly censusInstanceId: number;
+    public readonly censusMetagameEventType: MetagameEventType;
+    public state: Ps2alertsEventState;
+
+    constructor(
+        world: World,
+        timeStarted: Date,
+        timeEnded: Date | null,
+        zone: Zone,
+        censusInstanceId: number,
+        censusMetagameEventType: MetagameEventType,
+        state: Ps2alertsEventState,
+    ) {
+        super(
+            `${world}-${censusInstanceId}`, // lul
+            world,
+            timeStarted,
+            timeEnded,
+        );
+        this.zone = zone;
+        this.censusInstanceId = censusInstanceId;
+        this.censusMetagameEventType = censusMetagameEventType;
+        this.state = state;
+    }
+
+    public match(world: World, zone: Zone): boolean {
+        return world === this.world && zone === this.zone;
+    }
+
+    public overdue(): boolean {
+        // If now in milliseconds is greater th an start time + duration
+        return Date.now() > (this.timeStarted.getTime() + this.duration());
+    }
+
+    public duration(): number {
+        switch (this.censusMetagameEventType) {
```

Comment:
Map idea from before. Duration should probably be injected as a number or better as an Date object. That way we prevent the possibility that we have an event for which we cannot determine the duration after we have added it.

For instance now when taking the overdue instances the overdue thingy can crash because of one bad event.